### PR TITLE
[MAINT] Fixed wrong null-safety condition check in haicrypt

### LINKED ISSUE #2376: Either the condition 'NULL==crypto' is redundant or there is possible null pointer dereference: crypto.
https://github.com/Haivision/srt/blob/e8b713e954345291e8d6e2314d1e5094a6a14179/haicrypt/hcrypt_tx.c#L58
https://github.com/Haivision/srt/blob/e8b713e954345291e8d6e2314d1e5094a6a14179/haicrypt/hcrypt_tx.c#L86
https://github.com/Haivision/srt/blob/e8b713e954345291e8d6e2314d1e5094a6a14179/haicrypt/hcrypt_tx.c#L100
https://github.com/Haivision/srt/blob/e8b713e954345291e8d6e2314d1e5094a6a14179/haicrypt/hcrypt_tx.c#L133

### PR DESCRIPTION
**Pull request by ethouris**
Fixes #2376 

IMPORTANT: Since this version C99 is the minimum standard required for compiling haicrypt.

### CODE REVIEW

File: haicrypt/hcrypt_tx.c [hunk 1]
```diff
@@ -52,20 +52,21 @@ int HaiCrypt_Tx_GetBuf(HaiCrypt_Handle hhc, size_t data_len, unsigned char **in_
...
 	hcrypt_Session *crypto = (hcrypt_Session *)hhc;
-	hcrypt_Ctx *ctx = crypto->ctx;
-	int nbout = 0;
 
 	if ((NULL == crypto)
-	||  (NULL == ctx)
+	||  (NULL == crypto->ctx)
 	||  (NULL == out_p)
 	||  (NULL == out_len_p)) {
 		HCRYPT_LOG(LOG_ERR, "ManageKeys: invalid params: crypto=%p crypto->ctx=%p\n", crypto, ctx);
```
Comment (jeandube) [inline haicrypt/hcrypt_tx.c:63]: crypto may be NULL here and dereference ctx

File: haicrypt/hcrypt_tx.c [hunk 2]
```diff
@@ -83,28 +84,28 @@ int HaiCrypt_Tx_ManageKeys(HaiCrypt_Handle hhc, void *out_p[], size_t out_len_p[
 int HaiCrypt_Tx_GetKeyFlags(HaiCrypt_Handle hhc)
 {
 	hcrypt_Session *crypto = (hcrypt_Session *)hhc;
-	hcrypt_Ctx *ctx = crypto->ctx;
 
 	if ((NULL == crypto)
-	||  (NULL == ctx)){
+	||  (NULL == crypto->ctx)){
 		HCRYPT_LOG(LOG_ERR, "GetKeyFlags: invalid params: crypto=%p crypto->ctx=%p\n", crypto, ctx);
```
Comment (jeandube) [inline haicrypt/hcrypt_tx.c:90]: NULL crypto?

File: haicrypt/hcrypt_tx.c [hunk 3]
```diff
@@ -83,28 +84,28 @@ int HaiCrypt_Tx_ManageKeys(HaiCrypt_Handle hhc, void *out_p[], size_t out_len_p[
...
 	unsigned char *in_pfx, unsigned char *in_data, size_t in_len) 
 {
 	hcrypt_Session *crypto = (hcrypt_Session *)hhc;
-	hcrypt_Ctx *ctx = crypto->ctx;
-	int nbout = 0;
 
 	if ((NULL == crypto)
-	||  (NULL == ctx)){
+	||  (NULL == crypto->ctx)){
 		HCRYPT_LOG(LOG_ERR, "Tx_Data: invalid params: crypto=%p crypto->ctx=%p\n", crypto, ctx);
```
Comment (jeandube) [inline haicrypt/hcrypt_tx.c:103]: ditto

File: haicrypt/hcrypt_tx.c [hunk 4]
```diff
@@ -135,17 +138,18 @@ int HaiCrypt_Tx_Process(HaiCrypt_Handle hhc,
...
 	hcrypt_Session *crypto = (hcrypt_Session *)hhc;
-	hcrypt_Ctx *ctx = crypto->ctx;
-	int nb, nbout = 0;
 
 	if ((NULL == crypto)
-	||  (NULL == ctx)
+	||  (NULL == crypto->ctx)
 	||  (NULL == out_p)
 	||  (NULL == out_len_p)) {
 		HCRYPT_LOG(LOG_ERR, "Tx_Process: invalid params: crypto=%p crypto->ctx=%p\n", crypto, ctx);
```
Comment (jeandube) [inline haicrypt/hcrypt_tx.c:146]: ditto

File: haicrypt/hcrypt_tx.c [hunk 5]
```diff
@@ -135,17 +138,18 @@ int HaiCrypt_Tx_Process(HaiCrypt_Handle hhc,
...
 		return(-1);
 	}
 
+	hcrypt_Ctx *ctx = crypto->ctx;
```
Comment (jeandube) [inline haicrypt/hcrypt_tx.c:150]: reassigned line 156, which is redundant

File: haicrypt/hcrypt_tx.c [hunk 6]
```diff
@@ -52,20 +52,21 @@ int HaiCrypt_Tx_GetBuf(HaiCrypt_Handle hhc, size_t data_len, unsigned char **in_
...
 	if ((NULL == crypto)
-	||  (NULL == ctx)
+	||  (NULL == crypto->ctx)
 	||  (NULL == out_p)
 	||  (NULL == out_len_p)) {
 		HCRYPT_LOG(LOG_ERR, "ManageKeys: invalid params: crypto=%p crypto->ctx=%p\n", crypto, ctx);
 		return(-1);
 	}
 
+	hcrypt_Ctx *ctx = crypto->ctx;
```
Comment (jeandube) [inline haicrypt/hcrypt_tx.c:67]: reassigned line 73

**Review by jeandube — CHANGES_REQUESTED** [review]

File: haicrypt/hcrypt_tx.c (hunk 1 above)
Comment (ethouris) [reply haicrypt/hcrypt_tx.c:63]: No. The `NULL == crypto` condition is checked first, and if it's true, no other conditions are checked.

Comment (maxsharabayko) [reply haicrypt/hcrypt_tx.c:63]: `ctx` is not declared yet but used in the log.

Comment (ethouris) [reply haicrypt/hcrypt_tx.c:65]: Right, missed that. Should be fixed now.

File: haicrypt/hcrypt_tx.c [hunk 7]
```diff
@@ -149,11 +144,6 @@ int HaiCrypt_Tx_Process(HaiCrypt_Handle hhc,
 	/* Manage Key Material (refresh, announce, decommission) */
 	hcryptCtx_Tx_ManageKM(crypto);
```
Comment (ethouris) [inline haicrypt/hcrypt_tx.c:146]: @jeandube : Note that I'm taking it as a good deal that the call to `hcryptCtx_Tx_ManageKM` doesn't modify the assignment of `crypto->ctx`; as far as I could check it, it shouldn't.

**Review by maxsharabayko — APPROVED** [review]

**Comment by maxsharabayko** [comment]
Looks good to me.
@jeandube do you approve?

**Review by jeandube — APPROVED** [review]

**Pull request merged**